### ISSUE TITLE
Create more unit tests

### ISSUE THREAD

**Issue by horenbergerb** [horenbergerb/BernoulliDiffusion#7]
Self explanatory. I'd especially like to do more testing on the functions in model::BernoulliDiffusion.

Would be good to check that things like KL divergence are working as expected, verify that the beta values are properly calculated, etc.

Would be harder to check that sampling works as expected, but you could do it if you loaded some kind of dummy model into BernoulliDiffusion as the reverse process.

**Comment by horenbergerb** [comment]
I added a few tests for the beta values. It would be nice to have a few tests checking known cases of KL divergence and entropy

**Comment by horenbergerb** [comment]
Also, could be good to have a test for the whole training pipeline? Basically load some training data, feed it into the model, and check that the loss goes down?

**Comment by horenbergerb** [comment]
Ended up producing a lot more unit tests for math_utilities today, and they actually helped me catch some bugs.

The only major mathy stuff left to test would be the posterior and whatever else is in the forward loop of the model.

**Comment by horenbergerb** [comment]
Could also make some unit tests for the DataLoader and (if you're feeling ambitious) the plotting tools

**Comment by horenbergerb** [comment]
Note: I could make some tests for the posterior if I moved it into its own function. It's not too hard to calculate a few cases by hand and then compare to that
